bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LcnBindingProvider.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lcn;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.lcn.common.*;
+import org.openhab.binding.lcn.connection.*;
+import org.openhab.binding.lcn.input.*;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.transform.TransformationException;
+import org.openhab.core.transform.TransformationHelper;
+import org.openhab.core.transform.TransformationService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.TypeParser;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * Manages item-configurations.
+ * 
+ * @author Tobias J�ttner
+ */
+public class LcnBindingProvider extends AbstractGenericBindingProvider implements BindingProvider {
+	
+	/** Pattern used to parse bindings in general. */
+	private static final Pattern PATTERN_BINDING_GENERAL = Pattern.compile("(\\[.*?\\])*");
+	
+	/** Pattern used to parse bindings which are bound to openHAB commands. */
+	private static final Pattern PATTERN_BINDING_WITH_OPENHAB = Pattern.compile("\\[(.*?):(.*?):\'?(.*?)\'?\\]");
+	
+	/** Pattern used to parse bindings which don't have an openHAB command. */
+	private static final Pattern PATTERN_BINDING_PURE = Pattern.compile("\\[(.*):\'?(.*?)\'?\\]");
+	
+	/** Pattern to identify mappings in item bindings. */
+	private static final Pattern PATTERN_MAPPING = Pattern.compile("(.*?)\\((.*)\\)");
+	
+	/**
+	 * Stores all item names related to an LCN module address.
+	 * Used for optimization.
+	 * This map is never cleaned and might contain out-dated mappings or items that no longer exist.
+	 */ 
+	private final HashMap<LcnAddrMod, HashSet<String>> itemNamesByModulCache = new HashMap<LcnAddrMod, HashSet<String>>();
+	
+	/**
+	 * Item processing for the LCN bindings.
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item, String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
+		Matcher matcher = PATTERN_BINDING_GENERAL.matcher(bindingConfig);
+		if (!matcher.matches())
+			throw new BindingConfigParseException(bindingConfig + "' contains no valid binding!");
```

Comment:
since the Pattern is known you could give specific advice on how a configuration needs to look like in order to be valid